### PR TITLE
Search Bar UI & Functionality.

### PR DESCRIPTION
**Pull request by swoocn**
Submitting PR on behalf of @adisa39.
**[see commits]**

Update - it appears there was some confusion resulting in me accidentally submitting the PR for @hollusam changes.  Sorry about the premature PR!

### CODE REVIEW

**Comment by swoocn** [comment]
Great job @adisa39!

After consolidating your branch with /dev, I noticed that the search bar remains hidden once the map loads during local testing.  It seems like the z-index might need some adjustments.  Could you please take a look and make the necessary fixes? 
Thank you!

**Comment by swoocn** [comment]
Looks like CI/CD build failed while linting ->

image

**Comment by omotoshoayomikun** [comment]
Instead of using document.getElementById('search-input').focus() on line 31 of your SearchBar.tsx, you might want to consider using the useRef hook, which is more effective in React.

**Comment by hollusam** [comment]
> Instead of using document.getElementById('search-input').focus() on line 31 of your SearchBar.tsx, you might want to consider using the useRef hook, which is more effective in React.

Great, thanks for the suggestion, I'll get it sorted.

**Review by hollusam — APPROVED** [review]

**Review by swoocn — COMMENTED** [review]
LGTM! 😎👍
You have my approval; I would approve the PR but I can't since I'm the author.  Sorry about the mix up again!

File: src/components/shared/SearchBar/SearchBar.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+  const handleSearchBarChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setSearchBarValue(event.target.value);
+  };
+  
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleIconClick = () => {
+    // Move the placeholder text up
+    if(inputRef.current) {
```
Comment (swoocn) [inline src/components/shared/SearchBar/SearchBar.tsx:28]: `if (inputRef.current)` for consistency.

File: src/components/shared/SearchBar/SearchBar.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  const setSearchQuery = (query: string, searchType: string) => {
+    // setSearchQuery logic here
+  };
+  
+  const handleSubmitSearch = async (event: FormEvent) => {
+    event.preventDefault();
+    const query = searchBarValue;
+    if (query) {
+      const searchType = isBusinessSearchType ? 'business' : 'product';
+      console.info(`Search query emitted for ${searchType}: `, query);
```
Comment (swoocn) [inline src/components/shared/SearchBar/SearchBar.tsx:48]: We'll eventually need to replace these console logging with a logging framework; discussed **winston** earlier.

**Pull request merged**